lib/dor/text_extraction/file_fetcher.rb

```diff
@@ -80,6 +79,14 @@ def fetch_and_write_file_to_disk(filename:, path:)
       def preservation_client
         @preservation_client ||= Preservation::Client.configure(url: Settings.preservation_catalog.url, token: Settings.preservation_catalog.token)
       end
+
+      def aws_client
+        Aws.config.update({
```

Comment:
another option that wouldn't mess with global config would be to create a one-off `Resource` using the creds from settings, as is done in pres cat: https://github.com/sul-dlss/preservation_catalog/blob/c7b80c3f8ba9931dde1fccd8e846090ec002f94c/app/services/replication/aws_provider.rb#L28-L30

from which you can [get a `bucket` given a bucket name](https://github.com/sul-dlss/preservation_catalog/blob/c7b80c3f8ba9931dde1fccd8e846090ec002f94c/app/services/replication/aws_provider.rb#L18-L20), and then you can get an [`S3Object` from the bucket given the object key](https://github.com/sul-dlss/preservation_catalog/blob/c7b80c3f8ba9931dde1fccd8e846090ec002f94c/app/jobs/replication/delivery_job_base.rb#L16) (which can include slashes so that it's more path like, and i think slashes can be used for something that approximates directory listing, and other directory-ish affordances, but that is a very vague memory).  and then pres cat calls `upload_file` on the `S3Object`, but i actually don't see that in the API documentation?  and it seems like we want to try `write` for streaming anyway.

Reply:
idk if emulating pres cat is the right thing to do.  that code was written a while ago, and i'm not familiar enough with AWS development to know what's idiomatic.  but it seems like parts of that pattern (other than the `upload_file` call) could work well here, and consistency is nice if all else does turn out to be equal.

Reply:
Changing the interface/configuration to be more pres like for consistency.